crates/re_ui/src/list_item.rs

```diff
@@ -0,0 +1,196 @@
+    pub fn with_icon_fn(
+        mut self,
+        icon_fn: impl FnOnce(&ReUi, &mut egui::Ui, egui::Rect, egui::style::WidgetVisuals) + 'a,
+    ) -> Self {
+        self.icon_fn = Some(Box::new(icon_fn));
+        self
+    }
+
+    /// Provide a closure to display on-hover buttons on the right of the item.
+    pub fn with_buttons(
+        mut self,
+        buttons: impl FnOnce(&ReUi, &mut egui::Ui) -> egui::Response + 'a,
+    ) -> Self {
+        self.buttons_fn = Some(Box::new(buttons));
+        self
+    }
+
+    /// Draw the item.
+    pub fn show(self, ui: &mut Ui) -> Response {
+        ui.add(self)
+    }
+}
+
+impl egui::Widget for ListItem<'_> {
+    fn ui(self, ui: &mut Ui) -> Response {
+        let button_padding = ui.spacing().button_padding;
+        let icon_width_plus_padding = ReUi::small_icon_size().x + ReUi::text_to_icon_padding();
+        let icon_extra = if self.icon_fn.is_some() {
+            icon_width_plus_padding
+        } else {
+            0.0
+        };
+        let padding_extra = button_padding + button_padding;
+        let wrap_width = ui.available_width() - padding_extra.x - icon_extra;
+
+        let text =
+            self.text
+                .clone()
+                .into_galley(ui, Some(false), wrap_width, egui::TextStyle::Button);
+
+        let desired_size = (padding_extra + text.size() + egui::vec2(icon_extra, 0.0))
+            .at_least(egui::vec2(ui.available_width(), ReUi::list_item_height()));
+        let (rect, response) = ui.allocate_at_least(desired_size, egui::Sense::click());
+
+        response.widget_info(|| {
+            egui::WidgetInfo::selected(
+                egui::WidgetType::SelectableLabel,
+                self.selected,
+                text.text(),
+            )
+        });
+
+        if ui.is_rect_visible(rect) {
+            let visuals = if self.active {
+                ui.style().interact_selectable(&response, self.selected)
+            } else {
+                ui.visuals().widgets.inactive
+            };
+
+            let mut bg_rect = rect;
+            bg_rect.extend_with_x(ui.clip_rect().right());
+            bg_rect.extend_with_x(ui.clip_rect().left());
+            let background_frame = ui.painter().add(egui::Shape::Noop);
+
+            let min_pos = ui.painter().round_pos_to_pixels(egui::pos2(
+                rect.min.x.ceil(),
+                ((rect.min.y + rect.max.y - ReUi::small_icon_size().y) * 0.5).ceil(),
+            ));
+
+            // Draw icon
+            if let Some(icon_fn) = self.icon_fn {
+                let icon_rect = egui::Rect::from_min_size(min_pos, ReUi::small_icon_size());
+                icon_fn(self.re_ui, ui, icon_rect, visuals);
+            }
+
+            // Draw text next to the icon.
+            let mut text_rect = rect;
+            text_rect.min.x = min_pos.x + icon_extra;
+            let text_pos = ui
+                .layout()
+                .align_size_within_rect(text.size(), text_rect)
```

Comment:
This will follow the alignment of the current layout, but I think we should always align the label as left-center. You can use `Align2::LEFT_CENTER.align_size_within_rect` for that

Reply:
Oh, that maybe explains some of the behaviour I've seen with `CollapsibleState`.